tensorflow/compiler/xla/service/algebraic_simplifier_test.cc

```diff
@@ -5809,5 +5810,235 @@ TEST_F(AlgebraicSimplifierTest, SliceOfConcat) {
               GmockMatch(m::Parameter(1)));
 }
 
+TEST_F(AlgebraicSimplifierTest, SqrtOfSelfMultiply) {
+  const char* kModuleStr = R"(
+    HloModule m
+    test {
+      %p0 = f32[32]{0} parameter(0)
+      %multiply = f32[32]{0} multiply(f32[32]{0} %p0, f32[32]{0} %p0)
+      ROOT %sqrt = f32[32]{0} sqrt(f32[32]{0} %multiply)
+    }
+  )";
+  TF_ASSERT_OK_AND_ASSIGN(auto m, ParseAndReturnVerifiedModule(kModuleStr));
+  ASSERT_TRUE(AlgebraicSimplifier(default_options_).Run(m.get()).ValueOrDie());
+  EXPECT_THAT(m->entry_computation()->root_instruction(),
+              GmockMatch(m::Abs(m::Parameter(0))));
+}
+
+TEST_F(AlgebraicSimplifierTest, RsqrtOfRPower) {
```

Comment:
For a few tests you added that use the meta data, I think we need to add corresponding negative tests. The negative tests may not set the meta data, it may also just use a non-batch-norm related HLO value.

Reply:
I agree. Will do that.

Reply:
Btw do you want negative tests for all these tests? or would adding one or two cases suffice?

Reply:
I think two is good enough.

Reply:
The negative tests I added may a too detailed. Please let me know if this is what you expect or would just asserting that AlgebraicSimplifier change hasn't kicked in is enough. (`ASSERT_FALSE(AlgebraicSimplifier(default_options_).Run(m.get()).ValueOrDie());`)

Reply:
Checking the pass return false is good enough here.